icecreamshop/src/main/java/com/summerschool/icecreamshop/repository/ProductRepository.java

```diff
@@ -2,11 +2,16 @@
 
 import com.summerschool.icecreamshop.model.Product;
 import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.Query;
 import org.springframework.stereotype.Repository;
 
 import java.util.List;
 
 @Repository
 public interface ProductRepository extends JpaRepository<Product, Long> {
     List<Product> findById(int id);
+
+    @Query(nativeQuery = true, value =
+            "Select * From products Join (Select * From basket_products Where basket_products.basket_id = :basketId) as basket_products On products.id=basket_products.product_id")
+    List<Product> queryGetBasketProducts(Long basketId);
```

Comment:
Cum voi stii in cazul acesta care este cantitatea fiecarui produs din cos? Atributul `quantity` din `Product` probabil se refera la stoc si nu la numarul de produse de un anumit tip din cos.

Vad ca `BasketProduct` are un atribut `private Product product;`. Ai putea sa il incluzi si in DTO si iti va intoarce pentru fiecare produs din cos si detaliile produsului. Probabil va arata asa raspunsul:

```json
[{
   "id": int,
   "quantity": int,
   "product": {
        "description": string,
        etc...
    }
}]
```

Reply:
Daca incerc endpointul cu postman pare sa mearga si asa totusi 
![image](https://user-images.githubusercontent.com/62557131/133629158-382bd0bc-4fb3-4616-8699-28955a126fb6.png)

Nu stiu sigur dar cred ca ar putea fi acesata cantitatea din atributul basketProducts al produsului, nu?

Reply:
Poti pune tot raspunsul?

Reply:
[
    {
        "id": 1,
        "title": "Chocolate Mix Donuts",
        "shortDescription": "Treat yourself to ice cream infused with smooth Bourbon sprinkled",
        "longDescription": "Donut with chocolate icing, dark chocolate curls, Kinder Bueno™ wafers and delicate milk chocolate lines",
        "ingredients": [
            "Cream",
            "Milk",
            "Cane Sugar",
            "Cake Pieces",
            "Cake Base",
            "Sprinkles (Sugar, Coconut Oil, Buttermilk Powder, Natural Flavor, Spirulina Extract, Skim Milk)",
            "Eggs"
        ],
        "quantity": 100,
        "alergens": [
            "Milk",
            "Egg",
            "Soy",
            "Wheat"
        ],
        "price": 2.5,
        "currency": "USD",
        "photoUrlsBig": [
            "https://drive.google.com/file/d/1sp7d1GYCVy_Ss9E8ij2te1yC74HalBJg/view?usp=sharing"
        ],
        "photoUrlsMedium": [
            "https://drive.google.com/file/d/1Tk1Imi3-sRNzNeOLi6SuID8yUp-59BI8/view?usp=sharing"
        ],
        "photoUrlsSmall": [
            "https://drive.google.com/file/d/1TyUvmshHuIrUhHHpCmbaw9bBBNi6073_/view?usp=sharing"
        ],
        "type": "DONUTS",
        "rates": [
            {
                "id": 1,
                "rate": 5
            },
            {
                "id": 2,
                "rate": 4
            }
        ],
        "basketProduct": [
            {
                "id": 1,
                "quantity": 2
            }
        ]
    },
    {
        "id": 2,
        "title": "Chocolate Donuts",
        "shortDescription": "Treat yourself to ice cream infused with smooth Bourbon sprinkled",
        "longDescription": "Donut with chocolate icing",
        "ingredients": [
            "Cream",
            "Milk",
            "Cane Sugar",
            "Cake Pieces",
            "Cake Base",
            "Eggs"
        ],
        "quantity": 100,
        "alergens": [
            "Milk",
            "Egg",
            "Soy",
            "Wheat"
        ],
        "price": 2.5,
        "currency": "USD",
        "photoUrlsBig": [
            "https://drive.google.com/file/d/1hNy3z44jg434o72oeskkLnGLRstKvpVQ/view?usp=sharing"
        ],
        "photoUrlsMedium": [
            "https://drive.google.com/file/d/1vaXOOdJq-ttzcqWNY9rc4iWn-rZlByQG/view?usp=sharing"
        ],
        "photoUrlsSmall": [
            "https://drive.google.com/file/d/1wRWTl8EXIKX3c2oH5MKdVHn8w4sKq2Wm/view?usp=sharing"
        ],
        "type": "DONUTS",
        "rates": [
            {
                "id": 3,
                "rate": 5
            }
        ],
        "basketProduct": [
            {
                "id": 2,
                "quantity": 3
            }
        ]
    },
    {
        "id": 3,
        "title": "Strawberry Donuts",
        "shortDescription": "Treat yourself to ice cream infused with smooth Bourbon sprinkled",
        "longDescription": "Donut with strawberry icing and sprinkles",
        "ingredients": [
            "Cream",
            "Milk",
            "Cane Sugar",
            "Cake Pieces",
            "Cake Base",
            "Sprinkles (Sugar, Coconut Oil, Buttermilk Powder, Natural Flavor, Spirulina Extract, Skim Milk)",
            "Strawberry extract",
            "Eggs"
        ],
        "quantity": 100,
        "alergens": [
            "Milk",
            "Egg",
            "Soy",
            "Wheat"
        ],
        "price": 2.5,
        "currency": "USD",
        "photoUrlsBig": [
            "https://drive.google.com/file/d/1l02N3XRzkKlkSQbZzrgljiRjgdhxSBJ9/view?usp=sharing"
        ],
        "photoUrlsMedium": [
            "https://drive.google.com/file/d/1FruboQRUPHzmSxUhUE51mtQ14QnAoUVg/view?usp=sharing"
        ],
        "photoUrlsSmall": [
            "https://drive.google.com/file/d/1wbIT2F_5qaTTdyER4ixP5HPjPutzzcfl/view?usp=sharing"
        ],
        "type": "DONUTS",
        "rates": [
            {
                "id": 4,
                "rate": 5
            },
            {
                "id": 5,
                "rate": 2
            }
        ],
        "basketProduct": [
            {
                "id": 3,
                "quantity": 2
            }
        ]
    }
]

Reply:
asa arata tot raspunsul cand apelez http://localhost:8080/baskets/2

Reply:
Iti afiseaza doar datele produselor. E ca si cum ai apela `/products` doar ca verifici ca produsele sa fie in cos.